src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtPostcodeService.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.hmcts.dts.fact.services.admin;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.dts.fact.entity.Court;
+import uk.gov.hmcts.dts.fact.entity.CourtPostcode;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.repositories.CourtPostcodeRepository;
+import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class AdminCourtPostcodeService {
+    private final CourtRepository courtRepository;
+    private final CourtPostcodeRepository courtPostcodeRepository;
+
+    @Autowired
+    public AdminCourtPostcodeService(final CourtRepository courtRepository, final CourtPostcodeRepository courtPostcodeRepository) {
+        this.courtRepository = courtRepository;
+        this.courtPostcodeRepository = courtPostcodeRepository;
+    }
+
+    public List<String> getCourtPostcodesBySlug(final String slug) {
```

Comment:
This is an interesting one. Do we want to continue with returning the entire court and selecting a subset of it? 

At the moment it seems to be getting the entire court object, for us to only use a small portion of the data returned. 

Would it be more performant to use a PostcodeRepository or something to narrow down the search to one table? This would also potentially be better for adding it in. 

Bear in mind though that it could also be the case that getting the court is better for cascading purposes; specifically for deleting/adding, but probably not for getting